### PR TITLE
New variable card button

### PR DESCRIPTION
**Pull request by tealefristoe**
This PR makes some small changes to allow the CLUE tile toolbar new variable card button and zoom buttons to work properly.
- Adds a new DiagramHelper model that allows a client to zoom and position cards properly.
- Allows a client to hide several built in features of the diagram:
  - The new card button
  - The built-in zoom controls
  - The minimap/navigator
- Allows the client to disable interaction.

### CODE REVIEW

**Review by scytacki — CHANGES_REQUESTED** [review]
This looks good. I'm just hoping we can come up with a way to support the node positioning without exposing the react flow instance.

File: src/diagram/components/diagram.tsx [hunk 1]
```diff
@@ -27,15 +27,28 @@ const edgeTypes = {
 
 interface IProps {
   dqRoot: DQRootType;
+  externalReactFlowWrapper?: React.MutableRefObject<any>;
+  externalRfInstance?: any;
+  externalSetRfInstance?: (val: any) => void;
```
[the remark this answers is not in the document]
Comment (tealefristoe) [reply src/diagram/components/diagram.tsx:32]: It's used for `project` as well as the zoom functions. I'll see if I can figure out an alternative.

Do you think it's ok to keep the external react flow wrapper? It's only used for `getBoundingClientRect`.

**Review by scytacki — APPROVED** [review]
Seems good to me.

I don't think it needs to be an MST model it could just be a plain object. But we can change that later.

**Review by emcelroy — APPROVED** [review]
Looks good to me 👍

**Pull request merged**